### PR TITLE
MM-44806: Avoid pointers in JSON parsing in updateCategoryForTeamForUser

### PR DESCRIPTION
**Pull request by agnivade**
Passing null is a valid JSON input for a pointer struct. So we avoid
passing pointer to pointer.

https://mattermost.atlassian.net/browse/MM-44806
```release-note
NONE
```

### CODE REVIEW

**Review by noxer — APPROVED** [review]
LGTM

File: api4/channel_category_test.go [hunk 1]
```diff
@@ -417,6 +418,25 @@ func TestUpdateCategoryForTeamForUser(t *testing.T) {
...
+			require.Len(t, categories.Categories, 3)
+			require.Len(t, categories.Order, 3)
+
+			dmsCategory := categories.Categories[2]
+
+			payload := []byte(`null`)
+			route := fmt.Sprintf("/users/%s/teams/%s/channels/categories/%s", user.Id, th.BasicTeam.Id, dmsCategory.Id)
+			r, err := client.DoAPIPutBytes(route, payload)
+			require.Error(t, err)
+			defer closeBody(r)
```
Comment (noxer) [inline api4/channel_category_test.go:437]: Why is there a defer on the last line of the function?

**Review by noxer — CHANGES_REQUESTED** [review]

**Review by isacikgoz — APPROVED** [review]
LGTM, 0/5 on the `defer` statement. Not sure if it brings a significant overhead.

**Comment by noxer** [comment]
> 0/5 on the defer statement. Not sure if it brings a significant overhead.

It doesn't add much overhead, it doesn't help either. It's just unnecessary (and I'm a pedantic)

**Comment by agnivade** [comment]
I think Tim's point was more to the fact that it doesn't make sense to add a `defer` statement in the last line since it anyways will return after that point.

**Comment by isacikgoz** [comment]
> I think Tim's point was more to the fact that it doesn't make sense to add a `defer` statement in the last line since it anyways will return after that point.

Yeah I know, I was trying to say leaving it as is wouldn't bother me :) Leaving it to you guys.

**Review by noxer — APPROVED** [review]
LGTM, thank you for the change

**Pull request merged**